sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -160,20 +165,37 @@ export class BatchingReceiver extends MessageReceiver {
           // Setting drain must be accompanied by a flow call (aliased to addCredit in this case).
           this._receiver.drain = true;
           this._receiver.addCredit(1);
-        } else {
+
+          const drainTimeout = delay(this._drainTimeoutInMs).then(() => {
+            return Promise.reject();
```

Comment:
Just curious... what is the reasoning behind rejecting the promise here rather than resolving.

Reply:
Since I `reject` here, I can treat the timeout as an error and close the connection. I needed some way to determine who won the `Promise.race([drainPromise, drainTimeout])` so I can determine when to close, and since we treat timeouts as errors generally, this seemed like a reasonable way to branch that logic.

Note that I don't need `reject` to actually return an error since it never gets bubbled to the user anyway, it just determine whether we should close the link or not.